symengine/pow.cpp

```diff
@@ -94,7 +94,12 @@ RCP<const Basic> pow(const RCP<const Basic> &a, const RCP<const Basic> &b)
         return pownum(rcp_static_cast<const Number>(b), zero);
     }
     if (eq(*b, *one)) return a;
-    if (eq(*a, *zero)) return zero;
+    if (eq(*a, *zero) && is_a_Number(*b)) {
```

Comment:
No.
For `0**(positive number or zero)`, it is 0.
For `0**(negative number)`, it throws exception.
For others it simply returns `make_rcp<const Pow>(a, b)`.
Before it, we had `0**(anything) = 0`.

Sorry that I forgot to remove `0`**x` condition from `is_canonical`.